### ISSUE TITLE
software upgrade - helper app issue

### ISSUE THREAD

**Issue by efef** [eduvpn/macos#185]
when upgrading the client to the newest release, there is an helper app update issue.
When the new client is run and the helper app is being upgraded this error is shown:
<img width="399" alt="screen shot 2019-01-16 at 09 21 12" src="https://user-images.githubusercontent.com/3393474/51235563-1fad6680-1970-11e9-8720-e7b4ae1a0ff5.png">

**Comment by theNerdFromSiliconValley** [comment]
If we update the version number in OpenVPNHelper-Info.plist and  HelperService.swift file, SMJobless will update the helper file automatically.

**Comment by theNerdFromSiliconValley** [comment]
So during each release, we have to update the version no of the helper file

**Comment by efef** [comment]
Is it possible to make this a build.sh functionality so it automatically will increase?

**Comment by theNerdFromSiliconValley** [comment]
It is not recommended since we also edited on code side

**Comment by theNerdFromSiliconValley** [comment]
![5d597efa-3d1a-4b3f-9132-6c633af186a6](https://user-images.githubusercontent.com/39077747/51242641-ada64300-19aa-11e9-99c2-971d2f4b8858.jpeg)
Adding this is also recommened

**Comment by johankool** [comment]
> Is it possible to make this a build.sh functionality so it automatically will increase?

No, it is an error on the developer if this happens. The version needs to be increased when there have been changes made to the helper.

**Comment by johankool** [comment]
@theNerdFromSiliconValley You mention that the RunAtLoad key should be set to YES. Why is that? Did you read that somewhere?

**Comment by theNerdFromSiliconValley** [comment]
Yes, I saw and tested that DMG in @efef MAC ,

**Comment by johankool** [comment]
I don't understand your answer. Why do you think RunAtLoad should be added?

**Comment by theNerdFromSiliconValley** [comment]
@johankool 
Please check https://macadmins.psu.edu/files/2012/11/psumacconf2012-launchd.pdf

<img width="1400" alt="screen shot 2019-01-16 at 9 18 46 pm" src="https://user-images.githubusercontent.com/39077747/51258508-5b7a1780-19d4-11e9-900b-979b7d0e8cc5.png">

**Comment by johankool** [comment]
Afaict RunAtLoad causes the helper to be launched after boot, instead of on demand as is done now. Since we don't use it before the eduVPN app launches that doesn't add anything.

**Comment by theNerdFromSiliconValley** [comment]
Ok
